analyses/mwi/dryspells_trigger/15_mwi_hdx_processing.md

```diff
@@ -0,0 +1,138 @@
+## MWI processing for HDX
+
+Code to process MWI ARC2 rainy season and dry spell data frames for upload to HDX.
+
+```python
+import pandas as pd
+import os
+import fuzzyjoin
+import numpy as np
+
+data_dir = os.environ["AA_DATA_DIR"]
+
+rainy_df_path = os.path.join(data_dir, 'public', 'exploration', 'mwi', 'arc2', 'mwi_arc2_rainy_seasons.csv') 
+rainy_df = pd.read_csv(rainy_df_path)
+
+dry_spell_path = os.path.join(data_dir, 'public', 'exploration', 'mwi', 'arc2', 'mwi_arc2_centroid_dry_spells.csv')
+dry_spell_df = pd.read_csv(dry_spell_path, index_col=0, parse_dates=['dry_spell_last_date', 'dry_spell_first_date', 'dry_spell_confirmation'])
+
+daily_path = os.path.join(data_dir, 'public', 'exploration', 'mwi', 'arc2', 'mwi_arc2_precip_long_raw.csv')
+daily_df = pd.read_csv(daily_path, index_col=0, parse_dates=['date'])
+daily_df = daily_df[['ADM2_PCODE', 'date', 'mean_cell']]
+
+adm_path = os.path.join(data_dir, 'public', 'raw', 'mwi', 'cod_ab', 'mwi_district_lookup.xls')
+adm_df = pd.read_excel(adm_path)
+
+# Saving file paths
+
+RAINY_SEASON = os.path.join(data_dir, 'public', 'processed', 'mwi', 'arc2', 'mwi_arc2_rainy_seasons_2000_2020.csv')
+DRY_SPELLS = os.path.join(data_dir, 'public', 'processed', 'mwi', 'arc2', 'mwi_arc2_dry_spells_2000_2020.csv')
+```
+
+### Rainy season processing
+
+First, let's process the rainy season data.
+
+```python
+rainy_merged_df = pd.merge(rainy_df, adm_df[['ADM2_PCODE', 'ADM2_32EN', 'ADM1_EN']], left_on = 'pcode', right_on = 'ADM2_PCODE') \
+    .rename(columns={'ADM2_32EN':'ADM2_EN',
+                     'season':'season_approx',
+                     'ADM1_EN':'region'})
+                     
+rainy_merged_df['onset_date'] = pd.to_datetime(rainy_merged_df.rainy_season_onset)    
+rainy_merged_df['cessation_date'] = pd.to_datetime(rainy_merged_df.dry_season_first_date) - pd.to_timedelta(1,unit='d')
+rainy_merged_df['onset_month'] = rainy_merged_df.onset_date.dt.month
+rainy_merged_df['cessation_month'] = rainy_merged_df.cessation_date.dt.month
+rainy_merged_df['rainy_season_duration'] = (rainy_merged_df.cessation_date - rainy_merged_df.onset_date).dt.days
+
+# filter rows for seasons fully calculated
+rainy_filtered_df = rainy_merged_df[~rainy_merged_df.season_approx.isin(['1999-2000', '2020-2021'])]
+
+# get precipitation for rainy season
+
+ij_df = pd.merge(rainy_filtered_df, daily_df, how='inner', on='ADM2_PCODE')
+ij_df = ij_df[(ij_df.date >= ij_df.onset_date) & (ij_df.date <= ij_df.cessation_date)].reset_index()
+prec_df = ij_df.groupby(['ADM2_PCODE', 'season_approx'])['mean_cell'].sum().reset_index().rename(columns={"mean_cell":"rainy_season_rainfall"})
```

Comment:
Pretty detailed, but would round the `rainy_season_rainfall` column to a few decimals for readability

Reply:
Also the `cessation_month` and `rainy_season_duration` somehow end up being floats, would make them ints

Reply:
Will round. The other two are parsed as floats because there are `NaN` values and `int` types are not able to contain `NaN` apparently. Have moved filtration to be before calculations to correct this, although the resulting CSV is the same.

Reply:
To me it seems the csv is fine now :)